### PR TITLE
Document our monitoring setup

### PR DESCRIPTION
**Pull request by skitt**
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: src/content/monitoring/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+title = "Monitoring"
+date = 2020-08-12T16:02:00+02:00
+weight = 20
+pre = "<b>4. </b>"
++++
+
+## Basic Overview
+
+Submariner provides a number of Prometheus metrics, and sets up `ServiceMonitor` instances which allow these metrics to be scraped by an
+in-cluster Prometheus instance.
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:11]:
```suggestion
Submariner provides a number of metrics made available by `ServiceMonitor` instances that can consumed by an in-cluster Prometheus instance.
```

File: src/content/monitoring/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+The following metrics are exposed currently:
+
+* `submariner_gateways`, the number of gateways in the cluster;
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:15]:
```suggestion
* `submariner_gateways: the number of gateways in the cluster;
```

File: src/content/monitoring/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+* `submariner_connections`, the number of connections to other clusters, with the following labels:
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:17]:
```suggestion
* `submariner_connections`: the number of connections to other clusters, with the following labels:
```

File: src/content/monitoring/_index.en.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  * `local_cluster`, the local cluster name
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:19]:
```suggestion
  * `local_cluster`: the local cluster name
```

etc...

File: src/content/monitoring/_index.en.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+### OpenShift setup
+
+OpenShift 4.5 or later can automatically discover the Submariner metrics.
+This currently requires enabling user workload monitoring; see
+[the OpenShift documentation](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/monitoring/monitoring-your-own-services)
+for details.
+
+### Prometheus operator
+
+To start monitoring Submariner using the Prometheus operator, Prometheus needs to be configured to scrape the operator’s namespace
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:34]:
```suggestion
To start monitoring Submariner metrics using the Prometheus operator, Prometheus needs to be configured to access the Submariner namespace
```

File: src/content/monitoring/_index.en.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+(`submariner-operator` by default). The specifics depend on your Prometheus deployment, but typically, this will involve:
+
+* adding the operator’s namespace to Prometheus’ `ClusterRoleBinding`;
+
+* ensuring that Prometheus’ configuration doesn’t prevent it from scraping this namespace.
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:39]:
```suggestion
* ensuring that Prometheus’ configuration doesn’t prevent it from accessing this namespace.
```

File: src/content/monitoring/_index.en.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+This currently requires enabling user workload monitoring; see
+[the OpenShift documentation](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html/monitoring/monitoring-your-own-services)
+for details.
+
+### Prometheus operator
+
+To start monitoring Submariner using the Prometheus operator, Prometheus needs to be configured to scrape the operator’s namespace
+(`submariner-operator` by default). The specifics depend on your Prometheus deployment, but typically, this will involve:
+
+* adding the operator’s namespace to Prometheus’ `ClusterRoleBinding`;
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:37]:
```suggestion
* adding the Submariner namespace to Prometheus’ `ClusterRoleBinding`;
```

Comment (tpantelis) [inline src/content/monitoring/_index.en.md:37]:
```suggestion
* adding the Submariner namespace to Prometheus’ `ClusterRoleBinding`;
```

File: src/content/monitoring/_index.en.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+* ensuring that Prometheus’ configuration doesn’t prevent it from scraping this namespace.
+
+A minimal `Prometheus` object providing access to the Submariner metrics is as follows:
```
Comment (tpantelis) [inline src/content/monitoring/_index.en.md:41]: I assume the user must create the resource.

```suggestion
A `Prometheus` resource must be created to provide access to the Submariner metrics, for example:
```

Comment (mangelajo) [reply src/content/monitoring/_index.en.md:41]: I wonder if we could automate this from the operator too :)

**Review by mangelajo — CHANGES_REQUESTED** [review]
@skitt looks good, have a look to @tpantelis comments.

File: src/content/monitoring/_index.en.md (hunk 1 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:11]: The distinction is relevant IMO: Submariner provides metrics, and it also provides the configuration for an in-cluster Prometheus. The metrics can be used without the `ServiceMonitor`, and will be in many cases. “Scraping” is the appropriate term in Prometheus jargon.

File: src/content/monitoring/_index.en.md (hunk 2 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:15]: Done, thanks!

File: src/content/monitoring/_index.en.md (hunk 3 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:17]: Done, thanks!

File: src/content/monitoring/_index.en.md (hunk 4 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:19]: Done, thanks!

File: src/content/monitoring/_index.en.md (hunk 8 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:41]: I’ve rephrased the introductory paragraph to make it clear that the administrator needs to take care of both.

@mangelajo I considered that, but there are many different Prometheus deployments; we could try to figure out if the target is the default OpenShift setup, and configure that, but I don’t know what OpenShift “thinks” of tools changing its `ClusterRoleBinding`s behind its back ;-).

Comment (skitt) [reply src/content/monitoring/_index.en.md:42]: Duh, I forgot I documented OpenShift in the previous paragraph. The main issue here is that the `Prometheus` object specifies the overall Prometheus deployment; I think we should leave that up to the administrator...

Comment (mangelajo) [reply src/content/monitoring/_index.en.md:42]: I think we should eventually automate that, it makes sense, don't see why it should be a problem. We don't need to change it's ClusterRoleBinding, we can add the permissions to our own cluster roles

**Review by mangelajo — APPROVED** [review]

File: src/content/monitoring/_index.en.md (hunk 8 above)
Comment (skitt) [reply src/content/monitoring/_index.en.md:42]: Automating `Prometheus` setup is a bit like signing up with an ISP, and suddenly having workers show up to install an alarm and an extension to your home. I’m not sure how changing our own cluster roles helps, that wouldn’t give any extra permissions to Prometheus... (Prometheus pulls data; we don’t send data to Prometheus, it retrieves data from us.)

Comment (mangelajo) [reply src/content/monitoring/_index.en.md:42]: > Automating `Prometheus` setup is a bit like signing up with an ISP, and suddenly having workers show up to install an alarm and an extension to your home. I’m not sure how changing our own cluster roles helps, that wouldn’t give any extra permissions to Prometheus... (Prometheus pulls data; we don’t send data to Prometheus, it retrieves data from us.)

We could have a flag on ... if you want that done for you, or not... :)

**Comment by mangelajo** [comment]
@tpantelis looks good to me, can you have an eye to skitt comments ^ ?

**Pull request merged**